File: test-flakiness-detector/examples/basic.ts
/**
 * Basic usage example for Tool Name
 *
 * Run this example:
 *   npx tsx examples/basic.ts
 */

import { process } from '../src/index.js';

// Example 1: Basic usage
console.log('Example 1: Basic usage');
const result1 = process('hello world');
console.log('Input: "hello world"');
console.log('Output:', result1);
console.log();

// Example 2: With verbose option
console.log('Example 2: With verbose option');
const result2 = process('testing verbose mode', { verbose: true });
console.log('Input: "testing verbose mode"');
console.log('Output:', result2);
console.log();

// Example 3: Empty input
console.log('Example 3: Empty input');
const result3 = process('');
console.log('Input: ""');
console.log('Output:', result3);
console.log();

// Example 4: Special characters
console.log('Example 4: Special characters');
const result4 = process('Hello! @#$ 123');
console.log('Input: "Hello! @#$ 123"');
console.log('Output:', result4);

File: templates/rust-tool-template/examples/basic.rs
//! Basic usage example for tool-name
//!
//! Run this example:
//!   cargo run --example basic

use tool_name::{process, process_strict, Config};

fn main() {
    println!("Tool Name - Basic Usage Examples\n");

    // Example 1: Basic usage with default config
    println!("Example 1: Basic usage");
    let config = Config::default();
    let result = process("hello world", &config).unwrap();
    println!("  Input:  \"hello world\"");
    println!("  Output: \"{}\"\n", result);

    // Example 2: With verbose mode
    println!("Example 2: Verbose mode");
    let config = Config { verbose: true };
    let result = process("testing verbose", &config).unwrap();
    println!("  Output: \"{}\"\n", result);

    // Example 3: Empty input (allowed with process)
    println!("Example 3: Empty input with process()");
    let config = Config::default();
    let result = process("", &config).unwrap();
    println!("  Input:  \"\"");
    println!("  Output: \"{}\" (empty)\n", result);

    // Example 4: Empty input with strict mode (returns error)
    println!("Example 4: Empty input with process_strict()");
    let config = Config::default();
    match process_strict("", &config) {
        Ok(_) => println!("  Unexpected success"),
        Err(e) => println!("  Error: {}\n", e),
    }

    // Example 5: Unicode handling
    println!("Example 5: Unicode handling");
    let config = Config::default();
    let inputs = vec!["café", "über", "日本語"];
    for input in inputs {
        let result = process(input, &config).unwrap();
        println!("  \"{}\" -> \"{}\"", input, result);
    }

    println!("\nDone!");
}
